packages/client/components/DiscussionThreadInput.tsx

```diff
@@ -58,21 +60,38 @@ const DiscussionThreadInput = forwardRef((props: Props, ref: any) => {
     getMaxSortOrder,
     meeting,
     onSubmitCommentSuccess,
-    reflectionGroupId,
+    threadSourceId,
     threadParentId,
     replyMention,
     setReplyMention,
     dataCy
   } = props
   const isReply = !!props.isReply
   const isDisabled = !!props.isDisabled
-  const {id: meetingId, isAnonymousComment, teamId, viewerMeetingMember} = meeting
+  const {
+    id: meetingId,
+    isAnonymousComment,
+    teamId,
+    viewerMeetingMember,
+    meetingType
+  } = meeting
   const {user} = viewerMeetingMember
   const {picture} = user
   const [editorState, setEditorState] = useReplyEditorState(replyMention, setReplyMention)
   const atmosphere = useAtmosphere()
   const {submitting, onError, onCompleted, submitMutation} = useMutationProps()
   const placeholder = isAnonymousComment ? 'Comment anonymously' : 'Comment publicly'
+
+  let threadSource
+  switch (meetingType) {
```

Comment:
nit: in JS I've come to prefer object lookups vs. switches since they're a little more concise as the number of properties grows & from a DX perspective it means i can quickly make sure none of the cases will have side effects:
```
const threadSourceByMeetingType = {[MeetingTypeEnum.retrospective]: ThreadSourceEnum.REFLECTION_GROUP}
const threadSource = threadSourceByMeetingType[meetingType]
```

Reply:
thanks! i like that better too. change has been made.